core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala

```diff
@@ -81,15 +82,34 @@ import reactor.core.publisher.Flux
     }
   }
 
-  def updateOneInTx(stmt: Statement)(implicit ec: ExecutionContext): Future[Int] = {
+  final implicit class FluxOps[T](val flux: Flux[T]) extends AnyVal {
+
+    def toIndexedSeq: Mono[immutable.IndexedSeq[T]] = toIndexedSeq(identity)
+
+    def toIndexedSeq[U](map: T => U): Mono[immutable.IndexedSeq[U]] = {
+      val consumer: BiConsumer[mutable.Builder[U, IndexedSeq[U]], T] = (builder, elem) => builder += map(elem)
+      flux
+        .collect(() => IndexedSeq.newBuilder[U], consumer)
+        .map(_.result())
+    }
+  }
+
+  def updateOneInTx(stmt: Statement)(implicit ec: ExecutionContext): Future[Int] =
     stmt.execute().asFuture().flatMap { result =>
       result.getRowsUpdated.asFuture().map(_.intValue())(ExecutionContext.parasitic)
     }
-  }
+
+  def updateBatchInTx(stmt: Statement)(implicit ec: ExecutionContext): Future[Int] =
+    Flux
+      .from[Result](stmt.execute())
+      .concatMap(_.getRowsUpdated)
+      .toIndexedSeq(_.intValue())
+      .map(_.sum)
+      .asFuture()
```

Comment:
Not super important perhaps, but, could we collect/fold into sum to start with instead of into a seq and then sum?

Reply:
yes, we can. I will do it. 

Then we can also remove the FluxOps as it will be come obsolete

Reply:
Here 7da2f41